app/src/main/java/se/studieresan/studs/trip/TripFragment.kt

```diff
@@ -1,15 +1,259 @@
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ValueEventListener
 import se.studieresan.studs.R
+import se.studieresan.studs.StudsActivity
+import se.studieresan.studs.data.models.FeedItem
+import timber.log.Timber
 
-class TripFragment : Fragment() {
+private const val PERMISSION_FINE_LOCATION = 1
+private const val ERROR_MESSAGE_TEXT = "Location permission is required"
+
+class TripFragment : Fragment(), OnMapReadyCallback, OnFeedItemClickedListener, GoogleMap.OnPoiClickListener, GoogleMap.OnMarkerClickListener {
+
+    private lateinit var mapFragment: SupportMapFragment
+    private lateinit var fusedLocationClient: FusedLocationProviderClient
+    private lateinit var bottomSheetBehavior: BottomSheetBehavior<View>
+    private lateinit var feedRecyclerView: RecyclerView
+    private lateinit var adapter: FeedAdapter
+
+    private lateinit var listener: ValueEventListener
+    private lateinit var reference: DatabaseReference
+
+    private val userToMarkerMap = mutableMapOf<String, MarkerOptions>()
+    private var map: GoogleMap? = null
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
-        return inflater.inflate(R.layout.fragment_trip, container, false)
+        val view = inflater.inflate(R.layout.fragment_trip, container, false)
+        mapFragment = childFragmentManager.findFragmentById(R.id.map) as SupportMapFragment
+        mapFragment.getMapAsync(this)
+
+        adapter = FeedAdapter(this)
+
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(requireActivity())
+        showDeviceLocation()
+
+        bottomSheetBehavior = BottomSheetBehavior.from(view.findViewById(R.id.bottom_sheet_trip))
+        bottomSheetBehavior.run {
+            state = BottomSheetBehavior.STATE_COLLAPSED
+        }
+        view.findViewById<TextView>(R.id.tv_bottom_sheet_title).setOnClickListener {
+            if (bottomSheetBehavior.state == BottomSheetBehavior.STATE_EXPANDED) {
+                bottomSheetBehavior.state = BottomSheetBehavior.STATE_COLLAPSED
+            } else {
+                bottomSheetBehavior.state = BottomSheetBehavior.STATE_EXPANDED
+            }
+        }
+        feedRecyclerView = view.findViewById<RecyclerView>(R.id.rv_feed).apply {
+            layoutManager = LinearLayoutManager(requireContext())
+            adapter = this@TripFragment.adapter
+            addItemDecoration(DividerItemDecoration(requireContext(), (layoutManager as LinearLayoutManager).orientation))
+        }
+        setupDbListener()
+        view.findViewById<FloatingActionButton>(R.id.fab_share).setOnClickListener { showShareFragment() }
+        return view
+    }
+
+    private fun setupDbListener() {
+        val postLifetime = 60 * 60 * 24 // Posts disappear after 24 hours
```

Comment:
Is this just an UI thing or something related to Firebase?

Reply:
It only fetches the feed items from the last 24 hours.